src/python_testing/TC_VALCC_4_3.py

```diff
@@ -85,82 +79,43 @@ async def test_TC_VALCC_4_3(self):
         else:
             logging.info("Test step skipped")
 
-        self.step(4)
-        try:
-            await self.send_single_cmd(cmd=Clusters.Objects.ValveConfigurationAndControl.Commands.Open(openDuration=60), endpoint=endpoint)
-        except InteractionModelError as e:
-            asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
-            pass
-
-        self.step(5)
+        self.step("3a")
         utcTime = await self.read_single_attribute_check_success(endpoint=0, cluster=Clusters.Objects.TimeSynchronization, attribute=Clusters.TimeSynchronization.Attributes.UTCTime)
-        asserts.assert_true(utcTime is not NullValue, "OpenDuration is null")
-
-        self.step(6)
-        auto_close_time_dut = await self.read_valcc_attribute_expect_success(endpoint=endpoint, attribute=attributes.AutoCloseTime)
-
-        asserts.assert_true(auto_close_time_dut is not NullValue, "AutoCloseTime is null")
-        asserts.assert_greater_equal(auto_close_time_dut, (utcTime + 55000000),
-                                     "AutoCloseTime is not in the expected range")
-        asserts.assert_less_equal(auto_close_time_dut, (utcTime + 60000000), "AutoCloseTime is not in the expected range")
-
-        self.step(7)
-        try:
-            await self.send_single_cmd(cmd=Clusters.Objects.ValveConfigurationAndControl.Commands.Close(), endpoint=endpoint)
-        except InteractionModelError as e:
-            asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
-            pass
-
-        self.step(8)
-        auto_close_time_dut = await self.read_valcc_attribute_expect_success(endpoint=endpoint, attribute=attributes.AutoCloseTime)
-
-        asserts.assert_true(auto_close_time_dut is NullValue, "AutoCloseTime is not null")
 
-        self.step("9a")
-        defaultOpenDuration = await self.read_valcc_attribute_expect_success(endpoint=endpoint, attribute=attributes.OpenDuration)
+        self.step("3b")
+        if utcTime is not NullValue:
+            logging.info("UTCTime is not null, skipping test case")
 
-        asserts.assert_true(auto_close_time_dut is NullValue, "AutoCloseTime is not null")
+            # Skipping all remainig steps
+            for step in self.get_test_steps(self.current_test_info.name)[self.current_step_index:]:
```

Comment:
repeat of what we discussed offline, but if you just set the time, this should never come back null and if it does, it's a test failure.

Reply:
Yeah, I would not expect this to actually be null since my impression is also TH does indeed set UTCTime, but given we do have some specified behaviour for the case UTCTime is not set, we added this. I might need to find a way to actually get TH to NOT set UTCTime, but for now the intention is that having this in this form is better then not having it at all.